generic-sensor/generic-sensor-tests.js

```diff
@@ -0,0 +1,158 @@
+function runGenericSensorTests(sensorType, verifyReading) {
+  async_test(t => {
+    let sensor = new sensorType();
+    sensor.onchange = t.step_func_done(() => {
+      assert_true(verifyReading(sensor));
+      assert_equals(sensor.state, "activated");
+      sensor.stop();
+    });
+    sensor.onerror = t.step_func_done(event => {
+      assert_unreached(event.error.name + ":" + event.error.message);
+    });
+    sensor.start();
+  }, "event change fired");
+
+  async_test(t => {
+    let sensor1 = new sensorType();
+    let sensor2 = new sensorType();
+    sensor1.onactivate = t.step_func_done(() => {
+      // Reading values are correct for both sensors.
+      assert_true(verifyReading(sensor1));
+      assert_true(verifyReading(sensor2));
+
+      //After first sensor stops its reading values are null,
+      //reading values for the second sensor remains
+      sensor1.stop();
+      assert_true(verifyReading(sensor1, true /*should be null*/));
```

Comment:
Can you have two methods here instead?

e.g. assert_reading_not_null(sensor1) and  assert_reading_null(sensor1).

It's super hard to understand the test otherwise.